src/metrics.go

```diff
@@ -88,6 +87,20 @@ func (mc *metricsCollector) collect() {
 		go c.Collect(mc.db, &collectorWg, metricChan)
 	}
 
+	// Collect PDB metrics only when argument is set to 'PDB' or 'All'
+	collectPDBMetrics := args.SysMetricsSource == "PDB" || args.SysMetricsSource == "All"
+	if collectPDBMetrics {
+		collectorWg.Add(1)
+		go oraclePDBSysMetrics.Collect(mc.db, &collectorWg, metricChan)
+	}
+
+	// Collect Sys metrics by default and any value other than 'PDB'
+	collectSysMetrics := args.SysMetricsSource != "PDB"
```

Comment:
Any unexpected value (or any typo such as `Pdb`) in the new argument would lead to collecting sys_metrics only. This may be fine, but extending this argument behavior to support a different source would lead to potential breaking changes.

Did you consider simply defining a `bool` argument such as `args.PDBSysMetrics` (disabled by default) to enable the new PDB metrics?

Reply:
any other value would invoke the standard implementation and collect only the sys metrics. For people not bothered they can just move forward. Also string would allow the three combinations
1)  to collect only SYS
2) to collect only PDB
3) to collect SYS and PDB ( though  this one is mostly not required)

Reply:
that is actually a good point and consider making the checks as case insensitive

Reply:
Since collecting SYS and PDB is mostly not required, making it easier to switch between them using the string may be a better choice. We would need to include validation (and fail if the value is different from `''`, `'PDB'`, and `'ALL'`) or be very explicit in the argument description (and later on in the docs) to state that any value different from `PDB` and `ALL` will collect only SYS.

WDYT?

Reply:
Yes.. exactly my thought about having them in documentation. For any value other than 'PDB' and 'ALL' it will take the flow for argument being ''. Pushing changes for the checks to be case insensitive.